index.js

```diff
@@ -45,26 +33,30 @@ function lookup(addrs, opts, fn) {
     return fn(new Error('invalid or missing topic'), null);
   }
 
-  addrs.forEach(function(addr){
+  addrs.forEach(function(addr) {
     debug('lookup %s for topic %s', addr, opts.topic);
-    batch.push(function(done){
+    batch.push(function(done) {
       request
-      .get(addr + '/lookup?topic=' + opts.topic)
-      .timeout(timeout)
-      .retry(retries)
-      .end(function(err, res){
-        if (err) return done(err);
-        if (res.error) return done(res.error);
-        var data = res.body && res.body.data || {};
-        var producers = data.producers || [];
-        done(null, producers);
-      })
+        .get(addr + '/lookup?topic=' + opts.topic)
+        .timeout(timeout)
+        .retry(retries, (err, res) => {
+          if (res?.status === 500) {
+            return false;
+          }
```

Comment:
[`superagent-retry@0.5.1`](https://github.com/segmentio/nsq-lookup/pull/15/files#diff-7ae45ad102eab3b6d7e7896acd08c427a9b25b346470d7bc6507b6481575d519L15) does not [retry](https://github.com/segmentio/superagent-retry/blob/0.5.1/lib/retries.js) when the status code is 500. We're keeping that behaviour.

Reply:
`superagent@8.0.9` retries on the following status codes and errors: https://github.com/ladjs/superagent/blob/73c7efb5e9a0cf0409da9918d49b89055be29db5/src/request-base.js#L154-L178